cmd/opampsupervisor/supervisor/persistence.go

```diff
@@ -0,0 +1,92 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package supervisor
+
+import (
+	"crypto/rand"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"gopkg.in/yaml.v3"
+)
+
+// persistentState represents persistent state for the supervisor
+type persistentState struct {
+	InstanceID ulid.ULID `yaml:"instance_id"`
+
+	// Path to the config file that the state should be saved to.
+	// This is not marshaled.
+	configPath string `yaml:"-"`
+}
```

Comment:
This might be overkill to have this be a whole struct. We could get away with just writing the instance ID directly to a file instead of as marshaled YAML.

The advantage of doing it this way is that we can extend in the future to include more supervisor state if we need to add anything here. We can also version this state in the future if there are any changes to it (e.g. port the ULID representation over to arbitrary bytes representation, if we end up making that change)

Reply:
+1 to yaml.